File: ProductionPlanningSystem/ProductionPlanningSystem/src/main/java/com/example/ProductionPlanningSystem/EmployeeService.java
package com.example.ProductionPlanningSystem;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class EmployeeService {

    @Autowired
    private EmployeeRepository repository;

    public Employee addEmployee(Employee employee){
        return repository.save(employee);
    }

    public List<Employee> getEmployees(){
        return repository.findAll();
    }

    public Employee updateEmployee(Long id, Employee updated){

        Employee employee = repository.findById(id)
                .orElseThrow();

        employee.setEmployeeName(updated.getEmployeeName());
        employee.setDepartment(updated.getDepartment());
        employee.setDesignation(updated.getDesignation());
        employee.setEmail(updated.getEmail());
        employee.setPhone(updated.getPhone());

        return repository.save(employee);
    }

    public void deleteEmployee(Long id){
        repository.deleteById(id);
    }

}

File: react - production/planning/src/services/employeeService.js
import api from "../api/axios";

const getEmployees = async () => {
    const response = await api.get("/api/employees");
    return response.data;
};

const addEmployee = async (employee) => {
    const response = await api.post("/api/employees", employee);
    return response.data;
};

const updateEmployee = async (id, employee) => {
    const response = await api.put(`/api/employees/${id}`, employee);
    return response.data;
};

const deleteEmployee = async (id) => {
    await api.delete(`/api/employees/${id}`);
};

export default {
    getEmployees,
    addEmployee,
    updateEmployee,
    deleteEmployee
};